### ISSUE TITLE
eth_getLogs returns wrong blockHash

### ISSUE THREAD

**Issue by gzliudan** [XinFinOrg/XDPoSChain#208]
Now the rpc node of xinfin blockchain returns wrong blockHash when call method `eth_getLogs`. Below is an example:

**request:**
```shell
curl -s -X POST https://arpc.apothem.network/ -H "Content-Type: application/json" -d '
{
    "id": 1610,
    "jsonrpc": "2.0",
    "method": "eth_getLogs",
    "params": [
        {
            "address": "0x53350795c11cee781a7e174479778f848d76ab2a",
            "fromBlock": "0x22b2277",
            "toBlock": "0x22b2277",
            "topics": [
                [
                    "0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925",
                    "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
                    "0x6a12b3df6cba4203bd7fd06b816789f87de8c594299aed5717ae070fac781bac"
                ]
            ]
        }
    ]
}' | jq
```

**response:**
```json
{
  "jsonrpc": "2.0",
  "id": 1610,
  "result": [
    {
      "address": "xdc53350795c11cee781a7e174479778f848d76ab2a",
      "topics": [
        "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
        "0x0000000000000000000000000000000000000000000000000000000000000000",
        "0x000000000000000000000000ce77393582d39048a4fc73770ab19d2b239c45dd"
      ],
      "data": "0x000000000000000000000000000000000000000000084595161401484a000000",
      "blockNumber": "0x22b2277",
      "transactionHash": "0xbebc0bab1a0f9844b86b8edf3a169bb1eef9337ac8d2684d5ae758b6521c12fe",
      "transactionIndex": "0x1",
      "blockHash": "0x194a513ea40c94fccaca470cccb24c2ff3bd6398dec9a04e3eb3a4bfef8a330e",
      "logIndex": "0x2",
      "removed": false
    }
  ]
}
```

The right blockHash shoud be `0xe8ec0bfc021f839fd0a69ae89689e309632af2cc434e8a096732068f0736864b` according to https://explorer.apothem.network/blocks/0x22b2277#transactions. The blockHash `0x194a513ea40c94fccaca470cccb24c2ff3bd6398dec9a04e3eb3a4bfef8a330e` in response is wrong.

**Comment by simone1999** [comment]
I'm running into the same issues, which makes multiple applications I want to set up for the XDC network crash. The Block hashes actually seem random and are also not the parent hashes or comparable.

**Comment by gzliudan** [comment]
https://github.com/XinFinOrg/XDPoSChain/pull/171 will fix this issue.

**Comment by gzliudan** [comment]
gnosis safe use this api also.

**Comment by gzliudan** [comment]
It seems this bug disappeared from one block number. The blockHash of block number [45204373 is 0xeff3b01cb14bd7eaaaf3972249be2b69cea6724acb0949751c29c57bd675ab47](https://explorer.apothem.network/blocks/45204373#transactions)

**request**

```bash
curl -s -X POST https://erpc.apothem.network/  -H "Content-Type: application/json" -d '
{
    "id": 1610,
    "jsonrpc": "2.0",
    "method": "eth_getLogs",
    "params": [
        {
            "address": "0x53350795c11cee781a7e174479778f848d76ab2a",
            "fromBlock": "0x2B1C395",
            "toBlock": "0x2B1C395",
            "topics": [
                [
                    "0x8c5be1e5ebec7d5bd14f71427d1e84f3dd0314c0f7b2291e5b200ac8c7c3b925",
                    "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
                    "0x6a12b3df6cba4203bd7fd06b816789f87de8c594299aed5717ae070fac781bac"
                ]
            ]
        }
    ]
}' | jq
```

**response**

```bash
{
  "jsonrpc": "2.0",
  "id": 1610,
  "result": [
    {
      "address": "0x53350795c11cee781a7e174479778f848d76ab2a",
      "topics": [
        "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
        "0x000000000000000000000000d4ce02705041f04135f1949bc835c1fe0885513c",
        "0x00000000000000000000000085f33e1242d87a875301312bd4ebaee8876517ba"
      ],
      "data": "0x0000000000000000000000000000000000000000000000000de0b6b3a7640000",
      "blockNumber": "0x2b1c395",
      "transactionHash": "0x2a70ccf516712fd885dd1bf3e45cb6f0b75dd3dd7ad09c747b7798c081ccdca9",
      "transactionIndex": "0x1",
      "blockHash": "0xeff3b01cb14bd7eaaaf3972249be2b69cea6724acb0949751c29c57bd675ab47",
      "logIndex": "0x1",
      "removed": false
    }
  ]
}
```

**Comment by gzliudan** [comment]
Only the blockHash which is wrong in old blocks should be fixed now.

> if there is any way we can support this, let us know. This issue really would be great to get fixed as it's holding a lot of our developments back. We also run a mainnet full and archive node, so if access to it helps, I can generate a URL for you to access it.

**Comment by gzliudan** [comment]
Closed by #320.
